irohad/main/server_runner.cpp

```diff
@@ -71,3 +59,10 @@ void ServerRunner::shutdown() {
     serverInstance_->Shutdown();
   }
 }
+
+void ServerRunner::shutdown(
+    const std::chrono::system_clock::time_point &deadline) {
+  if (serverInstance_) {
+    serverInstance_->Shutdown(deadline);
+  }
```

Comment:
Might be write something at the else branch? It's not that normal flow, might be handy

Reply:
What will you do inside else branch?) there will be nothing to shut down or destroy in case of enclosing object was not fully initialized for some reason

Reply:
just logging :)